src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h

```diff
@@ -122,6 +122,12 @@ namespace OpenMS
       std::vector<MSSpectrum>& scored
     );
 
+    void extractSpectrum(
+      const String& experiment_path,
+      const String& target_list_path,
+      std::map<std::string,MSSpectrum>& transition_best_spec
```

Comment:
I am thinking that it might be best to return a FeatureMap class with annotations for each of the selected MSSpectrum with annotations for each of the annotated peaks.  The FeatureMap (essentially a container for Features) would contain Feature classes (containing information of the precursor mass, RT, and spectrum score) with subFeatures (describing the individual peaks and their attributes like mass, intensity, s/n etc.,).  This would require refactoring the scoreSpectrum class to return both a vector of MSSpectrum and FeatureMap.  The new class "selectSpectrum" would then sort and return the highest scoring MSSpectrum and associated FeatureMap similar to what you have now.

Reply:
Given the fact that other methods should also be modified, I'll merge this PR and open new issues to implement the changes you proposed